environment/src/main/java/com/sequenceiq/environment/environment/sync/EnvironmentStatusCheckerJob.java

```diff
@@ -0,0 +1,99 @@
+
+    private final FlowLogService flowLogService;
+
+    private final EnvironmentSyncService environmentSyncService;
+
+    private final EnvironmentStatusUpdateService environmentStatusUpdateService;
+
+    private final EnvironmentJobService environmentJobService;
+
+    public EnvironmentStatusCheckerJob(EnvironmentService environmentService, FlowLogService flowLogService,
+            EnvironmentSyncService environmentSyncService, EnvironmentStatusUpdateService environmentStatusUpdateService,
+            EnvironmentJobService environmentJobService) {
+        this.environmentService = environmentService;
+        this.flowLogService = flowLogService;
+        this.environmentSyncService = environmentSyncService;
+        this.environmentStatusUpdateService = environmentStatusUpdateService;
+        this.environmentJobService = environmentJobService;
+    }
+
+    @Override
+    protected void executeInternal(JobExecutionContext context) throws JobExecutionException {
+        Long envId = getEnvId();
+        Optional<Environment> environmentOpt = environmentService.findEnvironmentById(envId);
+        if (environmentOpt.isPresent()) {
+            Environment environment = environmentOpt.get();
+            prepareMdcContext(environment);
+            if (flowLogService.isOtherFlowRunning(envId)) {
+                LOGGER.debug("EnvironmentStatusCheckerJob cannot run, because flow is running for environment: {}", environment.getName());
+            } else {
+                syncAnEnv(environment);
+            }
+            MDCBuilder.cleanupMdc();
+        } else {
+            environmentJobService.unschedule(envId);
+            LOGGER.warn("EnvironmentStatusCheckerJob cannot run, because environment is not found with id: {}. This env is unscheduled now", envId);
+        }
+    }
+
+    @VisibleForTesting
+    void syncAnEnv(Environment environment) {
+        try {
+            ThreadBasedUserCrnProvider.doAs(environment.getCreator(), () -> {
+                EnvironmentStatus status = environmentSyncService.getStatusByFreeipa(environment);
+                if (environment.getStatus() != status) {
+                    if (!flowLogService.isOtherFlowRunning(environment.getId())) {
+                        environmentStatusUpdateService.updateEnvironmentStatusAndNotify(environment, status, ResourceEvent.ENVIRONMENT_SYNC_FINISHED);
+                    } else {
+                        LOGGER.info("EnvironmentStatusCheckerJob wants to update the status but it's ignored because a flow started on: {}",
+                                environment.getName());
+                    }
+                } else {
+                    LOGGER.debug("Environment status is the same ({}), the update is skipped", status);
+                }
+            });
+        } catch (Exception e) {
+            LOGGER.info("Environment sync is failed for {}, error: {}", environment.getName(), e.getMessage(), e);
```

Comment:
LOGGER.info signature is wrong, if you want to pass a Throwable then the first parameter is a String so you must format it.

Reply:
I've tried and working fine for me